Helm: add relevant pod annotations for certificate injecting via Vault Agent (#4660)
grafana/mimir: operations/helm/charts/mimir-distributed/CHANGELOG.md

* [CHANGE] Change number of Memcached max idle connections to 150. #4591
* [CHANGE] Set `unregister_on_shutdown` for `store-gateway` to `false` by default. #4690

> * [FEATURE] Add support for Vault Agent. When enabled, the pod annotations for TLS configurable components will be updated to allow a running Vault Agent to fetch secrets from Vault and inject them into the pod. The annotations will be updated for the following components: `admin-api`, `alertmanager`, `compactor`, `distributor`, `gateway`, `ingester`, `overrides-exporter`, `querier`, `query-frontend`, `query-scheduler`, `ruler`, `store-gateway`.

Reviewer: Don't forget to add the PR number at the end of the changelog item.
```suggestion
* [FEATURE] Add support for Vault Agent for the following components: `admin-api`, `alertmanager`, `compactor`, `distributor`, `gateway`, `ingester`, `overrides-exporter`, `querier`, `query-frontend`, `query-scheduler`, `ruler`, `store-gateway`. When enabled, the pod annotations for TLS configurable components will be updated to allow a running Vault Agent to fetch secrets from Vault and inject them into the pod. #4660
```

> * [FEATURE] Add support for Vault Agent. When enabled, the pod annotations for TLS configurable components will be updated to allow a running Vault Agent to fetch secrets from Vault and inject them into the pod. The annotations will be updated for the following components: `admin-api`, `alertmanager`, `compactor`, `distributor`, `gateway`, `ingester`, `overrides-exporter`, `querier`, `query-frontend`, `query-scheduler`, `ruler`, `store-gateway`. #4660

Reviewer 2: ```suggestion
* [FEATURE] Add support for Vault Agent. When enabled, the Pod annotations for TLS configurable components are updated to allow a running Vault Agent to fetch secrets from Vault and to inject them into a Pod. The annotations are updated for the following components: `admin-api`, `alertmanager`, `compactor`, `distributor`, `gateway`, `ingester`, `overrides-exporter`, `querier`, `query-frontend`, `query-scheduler`, `ruler`, `store-gateway`. #4660
```
